base/src/main/java/io/spine/code/js/FileName.java

```diff
@@ -64,62 +55,71 @@ private FileName(String value) {
+     *
+     * <p>All generated files have {@code _pb.js} suffix,
+     * so the descriptor with name {@code spine/options.proto}
+     * becomes {@code spine/options_pb.js}.
+     *
+     * <p>The resulting name also includes the path denoting a Protobuf package.
      *
      * @param descriptor
      *         the descriptor of the file
-     * @return the generated JavaScript file name
+     * @return the name of a generated JavaScript file
      */
-    public static FileName from(FileDescriptorProto descriptor) {
+    public static FileName from(FileDescriptor descriptor) {
         checkNotNull(descriptor);
         io.spine.code.proto.FileName protoFileName = io.spine.code.proto.FileName.from(descriptor);
-        String fileName = protoFileName.nameWithoutExtension() + SUFFIX + EXTENSION;
+        String fileName = protoFileName.nameWithoutExtension() + protoEnding();
         return of(fileName);
     }
 
     /**
-     * Obtains the file name from the passed descriptor.
+     * Composes the path from the given file to its root.
+     *
+     * <p>Basically, the method replaces all preceding path elements
+     * by the {@link FileReference#parentDirectory()}.
      */
-    public static FileName from(FileDescriptor descriptor) {
-        checkNotNull(descriptor);
-        return from(descriptor.toProto());
+    public String pathToRoot() {
+        List<String> pathElements = pathElements();
+        int fileLocationDepth = pathElements.size() - 1;
+        String pathToRoot = Strings.repeat(FileReference.parentDirectory(), fileLocationDepth);
+        String result = pathToRoot.isEmpty() ? FileReference.currentDirectory() : pathToRoot;
+        return result;
     }
 
     /**
-     * Returns all {@code FileName} elements, i.e. the relative path to the file and file name
-     * itself.
+     * Obtains the path to the file.
+     *
+     * <p>Assumes that the path is used at the root directory with generated Protobuf files.
      */
-    public String[] pathElements() {
-        String[] result = value().split(PATH_SEPARATOR);
-        return result;
+    public String pathFromRoot() {
+        return FileReference.currentDirectory() + value();
     }
 
     /**
-     * Composes the path from the given file to its root.
-     *
-     * <p>Basically, the method replaces all preceding path elements by the {@link #PARENT_DIR}.
+     * Returns all {@code FileName} elements, i.e. the relative path to the file and file name
+     * itself.
      */
-    public String pathToRoot() {
-        String[] pathElements = pathElements();
-        int fileLocationDepth = pathElements.length - 1;
-        String pathToRoot = Strings.repeat(PARENT_DIR, fileLocationDepth);
-        String result = pathToRoot.isEmpty() ? CURRENT_DIR : pathToRoot;
+    @VisibleForTesting
+    List<String> pathElements() {
+        List<String> result = Splitter.on(PATH_SEPARATOR)
+                                      .splitToList(value());
```

Comment:
Let's avoid using `@Beta` API here, since just `split` and create an immutable list is easy.